frontend/appflowy_tauri/src/appflowy_app/components/document/TextBlock/TextBlock.hooks.ts

```diff
@@ -105,10 +105,60 @@ function useTextBlockKeyEvent(id: string, editor: Editor) {
     };
   }, [keepSelection, backSpaceAction]);
 
+  const upEvent = useMemo(() => {
+    return {
+      key: keyBoardEventKeyMap.Up,
+      canHandle: canHandleUpKey,
+      handler: (...args: TextBlockKeyEventHandlerParams) => {
+        void focusPreLineAction({
+          editor: args[1],
+        });
+      },
+    };
+  }, [focusPreLineAction]);
+
+  const downEvent = useMemo(() => {
+    return {
+      key: keyBoardEventKeyMap.Down,
+      canHandle: canHandleDownKey,
+      handler: (...args: TextBlockKeyEventHandlerParams) => {
+        void focusNextLineAction({
+          editor: args[1],
+        });
+      },
+    };
+  }, [focusNextLineAction]);
+
+  const leftEvent = useMemo(() => {
+    return {
+      key: keyBoardEventKeyMap.Left,
+      canHandle: canHandleLeftKey,
+      handler: (...args: TextBlockKeyEventHandlerParams) => {
+        void focusPreLineAction({
+          editor: args[1],
+          focusEnd: true,
+        });
+      },
+    };
+  }, [focusPreLineAction]);
+
+  const rightEvent = useMemo(() => {
+    return {
+      key: keyBoardEventKeyMap.Right,
+      canHandle: canHandleRightKey,
+      handler: (...args: TextBlockKeyEventHandlerParams) => {
+        void focusNextLineAction({
```

Comment:
Why does both the `arrow.down` and `arrow.right` keys trigger the same handler `focusNextLineAction`? Shouldn't the `arrow.right key` trigger the something like `focusNextCharacter` handler instead?

Reply:
we handle these events reason is used to focus on another block. we did not handle these events when we just want to focus next char